r/tests/testthat/test-dataset.R

```diff
@@ -295,6 +295,29 @@ test_that("CSV dataset", {
   )
 })
 
+test_that("compressed CSV dataset", {
+  skip_if_not_available("gzip")
+  dst_dir <- make_temp_dir()
+  dst_file <- file.path(dst_dir, "data.csv.gz")
+  write.csv(df1, gzfile(dst_file), row.names = FALSE, quote = FALSE)
+  format <- FileFormat$create("csv", compression = CompressionType$GZIP)
```

Comment:
With the above suggestion, this becomes: 

```suggestion
  format <- FileFormat$create("csv", compression = "gzip")
```